csaf-import/src/main/kotlin/io/github/csaf/sbom/retrieval/RetrievedProvider.kt

```diff
@@ -19,28 +19,31 @@ package io.github.csaf.sbom.retrieval
 import io.github.csaf.sbom.retrieval.CsafLoader.Companion.lazyLoader
 import io.github.csaf.sbom.schema.generated.Provider
 import io.github.csaf.sbom.validation.ValidationContext
-import io.github.csaf.sbom.validation.ValidationFailed
 import io.github.csaf.sbom.validation.roles.CSAFProviderRole
 import io.github.csaf.sbom.validation.roles.CSAFPublisherRole
 import io.github.csaf.sbom.validation.roles.CSAFTrustedProviderRole
+import java.util.*
 import java.util.concurrent.CompletableFuture
-import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.SupervisorJob
+import java.util.stream.Stream
+import java.util.stream.StreamSupport
+import kotlinx.coroutines.*
+import kotlinx.coroutines.channels.ReceiveChannel
+import kotlinx.coroutines.channels.produce
+import kotlinx.coroutines.channels.toList
 import kotlinx.coroutines.future.future
 
 /**
  * This class represents a "retrieved" provider (i.e., the roles "publisher", "provider" and
  * "trusted provider"), including its metadata (in the form of [Provider]) as well as functionality
  * to retrieve its documents.
  */
-class RetrievedProvider(val json: Provider) {
+class RetrievedProvider(val json: Provider) : Validatable {
```

Comment:
I kind of liked that I got rid of the interface. Any particular reason why it was re-introduced?

Reply:
Yes, it enables sharing of the `validate()` function, which is exactly the same for `RetrievedAggregator` and `RetrievedProvider`.